### PR TITLE
Prohibit for loops which could be foreach with IntelliJ

### PR DESCRIPTION
**Pull request by leventov**
IntelliJ finds a few false positive occurrences (which are ignored using `//noinspection ForLoopReplaceableByForEach` for now) because of https://youtrack.jetbrains.com/issue/IDEA-177150

Also, removes `io.druid.collections.IntList`, because it is unused.

### CODE REVIEW

**Comment by drcrallen** [comment]
@leventov do you have any benchmarks on the impact of using the iterator instead of the older for loop methodology in some of these potentially tight loops?

File: processing/src/main/java/io/druid/segment/filter/Filters.java [hunk 1]
```diff
@@ -336,7 +336,7 @@ public static double estimateSelectivity(
   {
     long numMatchedRows = 0;
     for (int i = 0; i < bitmaps.size(); i++) {
-      final ImmutableBitmap bitmap = bitmapIndex.getBitmap(bitmaps.get(i));
+      final ImmutableBitmap bitmap = bitmapIndex.getBitmap(bitmaps.getInt(i));
```
Comment (drcrallen) [inline processing/src/main/java/io/druid/segment/filter/Filters.java]: unrelated change

File: processing/src/test/java/io/druid/query/aggregation/HistogramAggregatorTest.java [hunk 2]
```diff
@@ -108,14 +107,10 @@ public void testBufferAggregate() throws Exception
...
 
-    ArrayList<Float> b = Lists.newArrayList();
-    for (int i = 0; i < breaks.length; ++i) {
-      b.add(breaks[i]);
-    }
     HistogramAggregatorFactory factory = new HistogramAggregatorFactory(
         "billy",
         "billy",
-        b
+        Floats.asList(breaks)
```
Comment (drcrallen) [inline processing/src/test/java/io/druid/query/aggregation/HistogramAggregatorTest.java]: This is not the same functionality as the prior. The prior created an independent list instead of having the `breaks` be the backing array. is that intended?

**Comment by leventov** [comment]
@drcrallen the only potentially performance-sensitive place that is changed is `StringDimensionIndexer`, but there iteration over array (not `Iterable`) is replaced with for-each. For-each over an array translated into identical bytecode, as manual indexed loop.

File: processing/src/main/java/io/druid/segment/filter/Filters.java (hunk 1 above)
Comment (leventov) [inline processing/src/main/java/io/druid/segment/filter/Filters.java]: Without this change, the loop was a subject for the inspection. It could be replaced with `for (Integer bitmapIndex : bitmaps) { ... }`, but it uses boxing. Actually `get()` is deprecated in `IntList`.

File: processing/src/test/java/io/druid/query/aggregation/HistogramAggregatorTest.java (hunk 2 above)
Comment (leventov) [inline processing/src/test/java/io/druid/query/aggregation/HistogramAggregatorTest.java]: It is intended. Inside `HistogramAggregatorFactory` constructor, data is just copied from list back to an array.

File: common/src/test/java/io/druid/collections/IntListTest.java [hunk 3]
```diff
@@ -85,8 +85,8 @@ public void testToArray()
   {
     int[] inputArray = {1, 3, 4};
     IntList list = new IntList();
-    for (int i = 0; i < inputArray.length; i++) {
-      list.add(inputArray[i]);
+    for (int e : inputArray) {
```
Comment (jihoonson) [inline common/src/test/java/io/druid/collections/IntListTest.java]: Better to remove ```IntList``` because it's not used in anywhere.

Comment (leventov) [inline common/src/test/java/io/druid/collections/IntListTest.java]: Thanks, removed

File: processing/src/test/java/io/druid/query/groupby/epinephelinae/ByteBufferMinMaxOffsetHeapTest.java [hunk 4]
```diff
@@ -204,8 +204,8 @@ public void testRemove()
     ByteBuffer myBuffer = ByteBuffer.allocate(1000000);
     ByteBufferMinMaxOffsetHeap heap = new ByteBufferMinMaxOffsetHeap(myBuffer, limit, Ordering.<Integer>natural(), null);
 
-    for (int i = 0; i < values.size(); i++) {
-      heap.addOffset(values.get(i));
+    for (Integer value : values) {
```
Comment (drcrallen) [inline processing/src/test/java/io/druid/query/groupby/epinephelinae/ByteBufferMinMaxOffsetHeapTest.java]: Does this result in a lot of Integer boxing? or is the new java smart enough not to do that?

Comment (gianm) [inline processing/src/test/java/io/druid/query/groupby/epinephelinae/ByteBufferMinMaxOffsetHeapTest.java]: They were already boxed inside `values`.

Comment (leventov) [inline processing/src/test/java/io/druid/query/groupby/epinephelinae/ByteBufferMinMaxOffsetHeapTest.java]: It's iteration over `ArrayList<Integer>` so boxed already. Also it's in test so I didn't care about performance

Comment (drcrallen) [inline processing/src/test/java/io/druid/query/groupby/epinephelinae/ByteBufferMinMaxOffsetHeapTest.java]: Yeah, this question was more for my curiosity
